aws/src/main/java/org/apache/iceberg/aws/s3/S3Tag.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aws.s3;
+
+import java.util.Collection;
+import java.util.Map;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import software.amazon.awssdk.services.s3.model.Tag;
+
+public class S3Tag {
+
+  private static final String S3_FILE_IO_IMPL = "org.apache.iceberg.aws.s3.S3FileIO";
+
+  public S3Tag() {
+  }
+
+  public boolean isS3FileIOConfigured(Map<String, String> catalogProperties) {
+    String fileIOImpl = catalogProperties.get(CatalogProperties.FILE_IO_IMPL);
+    return fileIOImpl == null || fileIOImpl.equals(S3_FILE_IO_IMPL);
+  }
+
+  public Collection<Tag> toTags(TableIdentifier tableIdentifier) {
```

Comment:
I think we can hold on this feature later, let's first add `s3.tags.<key>=<val>` as a generic way to specify s3 tags, and in a subsequent PR we can add `s3.table-tags-enabled`

Reply:
Sounds good.